design/one-pager-resource-connectivity-mvp.md

```diff
@@ -0,0 +1,653 @@
+  [GKE cluster external resource]'s `.ipAllocationPolicy.useIpAliases` and
+  `.ipAllocationPolicy.createSubnetwork` fields to ensure an appropriate VPC
+  native subnet is automatically created and used.
+
+To add a generous amount of flexibility Crossplane could:
+
+* Add support for creating a new VPC network via a Crossplane VPC `Network`
+  managed resource.
+* Add support for creating a new VPC subnetwork via a VPC `Subnetwork` managed
+  resource.
+* Add support for `CloudSQLInstance` managed resources and their resource
+  classes to specify the external resource's `.ipConfiguration.privateNetwork`,
+  and specify the external name of a `Network` managed resource.
+* Add support for `GKECluster` managed resources to specify the [GKE cluster
+  external resource]'s `.network` and `.subnetwork` fields.
+* Add support for `GKECluster` managed resources and their resource classes to
+  specify the [GKE cluster external resource]'s `.network` field and
+  `.subnetwork` fields, and specify the external names of a `Network` and a
+  `Subnetwork` managed resource.
+* Use the `GKECluster` managed resource's existing support for setting the
+  [GKE cluster external resource]'s `.ipAllocationPolicy.useIpAliases` field.
+
+High-fidelity Crossplane managed resource representations of the aforementioned
+network and subnetwork external resources would look as follows:
+
+```yaml
+---
+apiVersion: vpc.gcp.crossplane.io/v1alpha1
+kind: Network
+metadata:
+  namespace: crossplane-system
+  name: example
+spec:
+  nameFormat: mycoolname
+  description: A really cool VPC network
+  # IPv4Range puts the network in subnetwork-less 'legacy mode'
+  IPv4Range: 10.0.0.0/8
+  # The documentation says this puts the VPC network in auto-create subnet mode.
+  # In my experiments it seemed to create the VPC in legacy mode instead.
+  autoCreateSubnetworks: true
+  peerings:
+  - name: peerwithsomeothercoolnetwork
+    network: someothercoolnetwork
+    autoCreateRoutes: true
+    exchangeSubnetRoutes: true
+  routingConfig:
+    routingMode: REGIONAL
+```
+
+```yaml
+---
+apiVersion: vpc.gcp.crossplane.io/v1alpha1
+kind: Subnetwork
+metadata:
+  namespace: crossplane-system
+  name: example
+spec:
+  nameFormat: mycoolname
+  description: My cool VPC subnetwork
+  network: projects/coolproject/global/networks/coolestvpc
+  ipCidrRange: 192.168.0.0/24
+  region: us-central-1
+  privateIpGoogleAccess: true
+  secondaryIpRanges:
+  - rangeName: pods
+    ipCidrRange: 10.0.0.0/8
+  - rangeName: services
+    ipCidrRange: 172.16.0.0/16
+  enableFlowLogs: true
+  logConfig:
+    enable: true
+    flowSampling: 0.5
+    metadata: INCLUDE_ALL_METADATA
+    aggregationInterval: 5-min
+```
+
+Putting this all together, the infrastructure administrator would configure the
+following to ensure that when an app operator created a `MySQLInstance` claim
+and `KubernetesCluster` claim the two would have connectivity:
```

Comment:
Maybe not relevant or perhaps my point of reference is outdated, but ... isn't `KubernetesCluster` resource a "privileged" resource type, i.e. created and maintained by the infrastructure administrator?

Reply:
`KubernetesCluster` is a resource claim. If we have made any distinction in which some resource claims are privilege and some are not, I'm unaware of it.